torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py

```diff
@@ -147,26 +164,46 @@ def _get_weight_obs_or_fq_ctr(quantization_config: Optional[QuantizationConfig])
         return None
     quantization_spec: QuantizationSpec = quantization_config.weight
     qdtype = _TORCH_DTYPE_TO_QDTYPE[quantization_spec.dtype]
-    if quantization_spec.qscheme == torch.per_tensor_symmetric:
-        return MinMaxObserver.with_args(
-            qscheme=quantization_spec.qscheme,
-            dtype=qdtype,
-            quant_min=quantization_spec.quant_min,
-            quant_max=quantization_spec.quant_max,
-            eps=2**-12,
-        )
-    elif quantization_spec.qscheme == torch.per_channel_symmetric:
-        return PerChannelMinMaxObserver.with_args(
-            qscheme=quantization_spec.qscheme,
-            dtype=qdtype,
-            quant_min=quantization_spec.quant_min,
-            quant_max=quantization_spec.quant_max,
-            eps=2**-12,
-        )
-    else:
-        raise Exception(
-            "Unsupported quantization_spec for weight: {}".format(quantization_spec)
-        )
+    if quantization_config.is_qat:
+        if quantization_spec.qscheme == torch.per_tensor_symmetric:
+            return FusedMovingAvgObsFakeQuantize.with_args(
+                observer=MovingAverageMinMaxObserver,
+                qscheme=quantization_spec.qscheme,
+                dtype=qdtype,
+                quant_min=quantization_spec.quant_min,
+                quant_max=quantization_spec.quant_max,
+                eps=2 ** -12,
+            )
+        elif quantization_spec.qscheme == torch.per_channel_symmetric:
+            return FusedMovingAvgObsFakeQuantize.with_args(
+                observer=MovingAveragePerChannelMinMaxObserver,
```

Comment:
Andrew, why does fake quantize have observer type specified here for per channel, unlike observer? Can this not be refactored to just take per channel arg?

Reply:
This is just following the existing implementation here: https://github.com/pytorch/pytorch/blob/0ebd3a78f483b7ea989276d2e29ef0ddce4ab153/torch/ao/quantization/fake_quantize.py#L465-L471

Not sure what you mean by "refactored to just take per channel arg"? You mean reducing the duplicate code in the two different qscheme cases?

Reply:
> You mean reducing the duplicate code in the two different qscheme cases

Yes. Immediately this is just about factoring out observer type.

Longer term is having obsever just take per_channel arg rather than having having separate PerChannel observer